FFMpegCore/FFMpeg/Arguments/IDynamicArgument.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Text;
+
+namespace FFMpegCore.Arguments
+{
+    public interface IDynamicArgument
+    {
+        /// <summary>
+        /// Same as <see cref="IArgument.Text"/>, but this receives the arguments generated before as parameter
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public string GetText(StringBuilder context);
```

Comment:
Change to take IArgument[] as input instead. That should also allow much more concise check ie. using .OfType<IInputArgument>().Any() or similar. Fits together with me comment below